Project21/Project21/Palindrome.c

```diff
@@ -0,0 +1,153 @@
+   }
+}
+
+/// <summary>Function to print result</summary>
+void PrintResult (bool value) {
+   printf (value ? ("\nIt is a Palindrome\n\n") : ("\nIt is not a Palindrome\n\n"));
+}
+
+/// <summary>Clear the content on screen</summary>
+void ClrScreen () {
+#ifdef _WIN32
+   system ("cls");
+#endif 
+}
+
+/// <summary>Function to check the given number is palindrome or not </summary>
+bool NumberIsPalindrome (int num) {
+   int org = num;
+   long long int rev = 0; // used to store reverse number exceeding integer range
+   while (num != 0) {
+      rev = rev * 10 + num % 10;
+      num /= 10;
+   }
+   if (org < 0) {
+      printf ("Output: %-7lld", rev);
+      rev = -rev;
+   }
+   else printf ("Output: %-7lld", rev);
+   return (rev == org);
+}
+
+/// <summary>Function to check the given string is palindrome or not </summary>
+bool StringIsPalindrome (char* sentences) {
+   int i = 0, j = 0, length = (int)strlen (sentences);
+   char temp[MAX_CHAR];
+   while (i < length) {
+      if (isalnum (sentences[i]) != 0)
+         temp[j++] = sentences[i];
+      i++;
+   }
+   temp[j] = '\0';
+   if (j == 0) return false;
+   for (int n = 0; n < (j / 2); n++) {
+      if (toupper (temp[n]) != toupper (temp[j - n - 1])) return false;
+   }
+   return true;
+}
+
+/// <summary>Function to get integer input from user</summary>
+void IntegerInput (int option) {
+   errno = 0;
+   char input[MAX_VALUE], * endptr, * intResult, * i;
+   printf ("\nEnter an integer: ");
+   i = fgets (input, sizeof (input), stdin);
+   intResult = strchr (input, '\n');
+   int number = strtol (input, &endptr, 10);
+   if (intResult == NULL) ClrBuffer (option);
+   if (i == NULL || endptr == input || *endptr != '\n' || input[0] == ' ' || input[0] == '\t') printf ("INVALID!!!\n\n");
+   else if (errno == ERANGE) printf ("INPUT LIMIT EXCEEDED\n");
+   else PrintResult (NumberIsPalindrome (number));
+}
+
+/// <summary>Function to get string input from user</summary>
+void StringInput (int option) {
+   char sentences[MAX_CHAR], * j, * strResult;
+   printf ("\nEnter a sentence: ");
+   j = fgets (sentences, sizeof (sentences), stdin);
+   strResult = strchr (sentences, '\n');
+   if (j != NULL && sentences[0] != '\n' && strResult != NULL)
+      PrintResult (StringIsPalindrome (sentences));
+   else {
+      printf ("INVALID!!!\n\n");
+      ClrBuffer (option);
+   }
+}
+
+/// <summary>Function to check test cases</summary>
+void OutputCheck () {
+   printf ("\n*****PALINDROME CHECK*****\n");
+   char* strInput[] = { "Don't nod","Able was I ere I saw Elba", "racecar","Otto","Trumpf Metamation","RADAR","NASA","MALAYALAM","Kayak","LEVEL" };
+   int strExpected[] = { 1,1,1,1,0,1,0,1,1,1 };
+   int strLength = sizeof (strInput) / sizeof (strInput[0]);
+   for (int i = 0; i < strLength; i++) {
+      int result = StringIsPalindrome (strInput[i]);
+      printf ("Test Case%3d: Input--> %-28s ", i + 1, strInput[i]);
+      printf ((result == strExpected[i]) ? " PASS\n" : " FAIL\n");

```

Comment:
Give proper prompt, whether you're checking the output of the input or the test case passed the test